Automattic/wp-calypso
# Performance Profiler: weekly report subscribe page

Fixes https://github.com/Automattic/dotcom-forge/issues/8823

## Proposed Changes

* Add a new `/speed-test-tool/weekly-report` URL
* Make the site-build re-usable for the 404 page later (Iix1FFHIQdfECIXn5JKT5s-fi-4986_87188)
* On page load, send a request to the `/lead` endpoint to subscribe the user to weekly updates

## Why are these changes being made?


https://github.com/Automattic/dotcom-forge/issues/8823

## Testing Instructions



### Logged-in
* Go to `/speed-test` and start a new test
* In the Calypso environment, copy over the finished test results eg. `/speed-test-tool?url=https%3A%2F%2Fwordpress.com%2F&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NzA1N30.7KXaBL1qcm0y0GqvM8ymJ0pbST9ls6P09_hri8EaKSI`
* Open the devtools, and go to the Network tab
* On the loaded test page, scroll down to the weekly report signup section, and click the Sign Up button
* Check that the site is redirected to the `/speed-test-tool/weekly-report` page
* On the Network tab, check that the POST request to the `/lead` endpoint is sent
* Test the loading and error state by blocking the `lead` request or modifying the flags in the `weekly-report/index.tsx` file

| Loading | Successful request | Error |
|--------|--------|--------|
| ![CleanShot 2024-09-02 at 12 49 23@2x](https://github.com/user-attachments/assets/7a2b31d6-93ba-45d4-be2c-b6f585f3929b) | ![CleanShot 2024-09-02 at 12 48 41@2x](https://github.com/user-attachments/assets/4eea71f8-295e-490c-8380-48fd96fa6149) | ![CleanShot 2024-09-02 at 12 49 54@2x](https://github.com/user-attachments/assets/16e5d6f7-05e4-4b1e-b968-5a324f2922dc) | 

> [!NOTE]
> The log-in redirection can't be tested on the Calypso environment because of the environment mismatch when clicking the log-in link from an email. After deploying this PR, we will have to test the logged-out flow in production.

### Logged-out
* Go to `/speed-test` and start a new test
* In the Calypso environment, copy over the finished test results eg. `/speed-test-tool?url=https%3A%2F%2Fwordpress.com%2F&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NzA1N30.7KXaBL1qcm0y0GqvM8ymJ0pbST9ls6P09_hri8EaKSI`
* Open the devtools, and go to the Network tab
* On the loaded test page, scroll down to the weekly report signup section, and click the Sign Up button
* Check that you're redirected to the `/log-in` page and that the `redirect_to` parameter preserves the query params
* Enter a test email address and create a new account
* Check that you're redirected to the success page and that the POST request to the `/lead` endpoint is sent

## Pre-merge Checklist
